docs/reference/userapi.rst

```diff
@@ -70,3 +76,129 @@ that best suits that file-format.
+
+If you have an old script that you do not wish to migrate, you can avoid most of 
+the migration by explicitly importing the v2 API::
+
+    import imageio.v2 as imageio
+
+This will give you access to most of the old API. However, calls will still rely
+on the new (v3) plugins and backends, which may cause behavioral changes. As
+such, you should prefer a full migration to v3 and should avoid using the v2 API
+in new code. This is primarily as a quick way for you to postpone a full
+migration until it is convenient for you to do so.
+  
+
+Reading Images 
+--------------
+
+`iio.imread` can now return a ndimage instead of being limited to flat images.
+As such, `iio.volread` has merged into `iio.imread` and is now gone. Similarily,
+`iio.mimread` and `iio.mvolread` have merged into a new function called
+`iio.imiter`, which returns a generator that yields ndimages from a file in the
+order in which they appear. Further, the default behavior of `iio.imread` has
+changed, and it now returns the first ndimage (instead of the first flat image)
+if the image contains multiple images.
+
+To reproduce similar behavior to the v2 API behavior use one of the following
+snippets::
+
+    # img = iio.v2.imread(image_resource)
+    try:
+        img = iio.v3.imread(image_resource, index=None)[0]
+    except ValueError:
+        img = iio.v3.imread(image_resource, index=0)
+
+    # img = iio.v2.volread(image_resource)
+    img = iio.v3.imread(image_resource)
+    
+    # img = iio.v2.mimread(image_resource)
+    # img = iio.v2.mvolread(image_resource)
+    img = [im for im in iio.v3.imiter(image_resource)]
+
+Writing Images
+--------------
+
+Similar to reading images, the new `iio.imwrite` can handle ndimages.
+``iio.mimwrite``, ``iio.volwrite``, and ``iio.mvolwrite`` have all dissapeared.
+The same goes for their aliases ``iio.mimsave``, ``iio.volsave``,
+``iio.mvolsave``, and ``iio.imsave``. They are now all covered by
+``iio.imwrite``.
+
+To reproduce similar behavior to the v2 API behavior use one of the following
+snippets::
+
+    # img = iio.v2.imwrite(image_resource, ndimage)
+    # img = iio.v2.volwrite(image_resource, ndimage)
+    img = iio.v3.imwrite(image_resource, ndimage)
+
+    # img = iio.v2.mimwrite(image_resource, list_of_ndimages)
+    # img = iio.v2.mvolwrite(image_resource, list_of_ndimages)
+    img = iio.v3.imwrite(image_resource, list_of_ndimages)
+
+Reader and Writer
+-----------------
+
+Plugins now longer have nested ``Reader`` and ``Writer`` classes and they are no
+longer global singletons. Instead, a new plugin object is instantiated in mode ``r``
+(reading) or  ``w`` (writing) for each `ImageResource` being opened.
+As such, the functions ``iio.get_reader``, ``iio.get_writer``, ``iio.read``, and
+``iio.save`` have become obsolete and the plugin object is used directly
+instead. Using these is considered advanced usage, happens inside a context
+manager, and uses `iio.imopen`::
```

Comment:
The mentioning of nested classes and singletons is not relevant for normal users, so let's omit that.

```suggestion
Previously, the `reader` and `writer` objects provided an advanced way to read/write data with more control. These are now removed. Likewise, the functions ``iio.get_reader``, ``iio.get_writer``, ``iio.read``, and
``iio.save`` have become obsolete.

Instead, the plugin object is used directly, which is instantiated in mode ``r`` (reading) or  ``w`` (writing).
To create a plugin object, call `iio.imopen` and use it as a context manager::
```